# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerService.h
# Commit: 1e774bd7f365
# Full Hash: 1e774bd7f36525a03b6260b1cd7741bcf4e5f622
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-09 04:12:49
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r b003e26209cf -r 1e774bd7f365 dom/workers/remoteworkers/RemoteWorkerService.h
--- a/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Apr 08 09:43:33 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Apr 08 09:43:33 2025 +0000
@@ -13,12 +13,14 @@
 #include "nsCOMPtr.h"
 #include "nsIObserver.h"
 #include "nsISupportsImpl.h"
+#include "mozilla/dom/RemoteWorkerTypes.h"
 
 class nsIThread;
 
 namespace mozilla::dom {
 
 class RemoteWorkerDebuggerManagerChild;
+class RemoteWorkerDebuggerManagerParent;
 class RemoteWorkerService;
 class RemoteWorkerServiceChild;
 class RemoteWorkerServiceShutdownBlocker;
@@ -123,7 +125,8 @@
 
   nsCOMPtr<nsIThread> mThread;
   RefPtr<RemoteWorkerServiceChild> mActor;
-  RefPtr<RemoteWorkerDebuggerManagerChild> mDebuggerManagerActor;
+  RefPtr<RemoteWorkerDebuggerManagerChild> mDebuggerManagerChild;
+  RefPtr<RemoteWorkerDebuggerManagerParent> mDebuggerManagerParent;
   // The keep-alive is set and cleared on the main thread but we will hand out
   // additional references to it from the "Worker Launcher" thread, so it's
   // appropriate to use a mutex.  (Alternately we could have used a ThreadBound